gdk4/src/texture.rs

```diff
@@ -18,3 +18,175 @@ impl<O: IsA<Texture>> TextureExtManual for O {
         }
     }
 }
+
+impl Texture {
+    pub async fn for_pixbuf_future(pixbuf: &gdk_pixbuf::Pixbuf) -> Texture {
+        let (sender, receiver) = futures_channel::oneshot::channel();
+
+        let task = gio::Task::new(
+            Some(&pixbuf.upcast_ref()),
+            gio::Cancellable::NONE,
+            move |result, _| {
+                let texture = result
+                    .downcast_ref::<gio::Task>()
+                    .unwrap()
+                    .propagate_value()
+                    .unwrap()
+                    .get::<Texture>()
+                    .unwrap();
+                let _result = sender.send(texture);
+            },
+        );
+
+        let task_func = move |task: &gio::Task,
+                              source_object: Option<&glib::Object>,
+                              _: Option<&gio::Cancellable>| {
+            let pixbuf = source_object
+                .unwrap()
+                .downcast_ref::<gdk_pixbuf::Pixbuf>()
+                .unwrap();
+            let texture = Texture::for_pixbuf(&pixbuf);
+            task.return_value(&texture.to_value());
+        };
+
+        task.run_in_thread(task_func);
+
+        let texture = receiver.await.unwrap();
+
+        texture
+    }
+
+    #[cfg(any(feature = "v4_6", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v4_6")))]
+    pub async fn from_bytes_future(bytes: &glib::Bytes) -> Result<Texture, glib::Error> {
+        let (sender, receiver) = futures_channel::oneshot::channel();
+
+        let task = gio::Task::new(
+            None::<&glib::Object>,
+            gio::Cancellable::NONE,
+            move |result, _| {
+                let texture_result = result
+                    .downcast_ref::<gio::Task>()
+                    .unwrap()
+                    .propagate_value()
+                    .map(|value| value.get::<Texture>().unwrap());
+                let _result = sender.send(texture_result);
+            },
+        );
+
+        let task_func =
+            move |task: &gio::Task, _: Option<&glib::Object>, _: Option<&gio::Cancellable>| {
+                let result = Texture::from_bytes(&bytes);
+                task.return_value(&result.to_value());
+            };
+
+        task.run_in_thread(task_func);
+
+        let texture_result = receiver.await.unwrap();
+
+        texture_result
+    }
+
+    pub async fn from_file_future(file: &impl IsA<gio::File>) -> Result<Texture, glib::Error> {
+        let (sender, receiver) = futures_channel::oneshot::channel();
+
+        let task = gio::Task::new(
+            Some(file.as_ref().upcast_ref()),
+            gio::Cancellable::NONE,
```

Comment:
Is it possible to make them cancellable? I might need that.

Reply:
Probably, by exposing the GTask's cancellable as param. But then, not sure how you would cancel the ongoing operation..

Reply:
Yeah, I don't think the operation can be cancelled, but I guess it could at least return an error when the operation is done, which is still somewhat useful imho.